### ISSUE TITLE
GitHub Actions credentials appear to expire early for benchmark runs exceeding 1 hour

### ISSUE THREAD

**Issue by sauraank** [awslabs/mountpoint-s3#784]
### Mountpoint for Amazon S3 version

mountpoint-s3 v1.4.1

### AWS Region

us-east-1

### Describe the running environment

Running on Self hosted runner for Github Actions.
Operating system:
Distribution: Ubuntu 22.04.3 LTS
Architecture: x86_64
EC2 instance:
Instance type: m5n.24xlarge
Availability zone: us-east-1b

### Mountpoint options

```shell
target/release/mount-s3 BUCKET-NAME /tmp/mount-s3-g6N4H9xvWgzl --debug --allow-delete --log-directory=logs/readdir_100000 --prefix=benchmark/
```

### What happened?

Github OIDC assume IAM role (i.e. ask AWS STS for a set of short-term credentials) and use them for running the github action. When a bucket is mounted by mountpoint using Assume Role and workload is run for longer than an hour, it starts throwing `ExpiredToken` error. It might be happening with other Web Identity which Assume Role as well. 

### Relevant log output

```shell
Our benchmark logs: https://github.com/awslabs/mountpoint-s3/actions/runs/8002443498/job/21855663125#step:8:190.
```

**Comment by sauraank** [comment]
We can probably increase the session duration to upto [12 hours](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html#API_AssumeRole_RequestParameters).
But, I think we would like the credentials to refreshed after 1 session and not throw error?

**Comment by dannycjones** [comment]
Just sharing what we discussed this morning together here:

Our CI is getting a set of session credentials from `aws-actions/configure-aws-credentials` and populating this in the environment for the job. We don't expect those credentials to be refreshed, as they are static environment credentials and it appears to be behave as expected if - for example - you used short-term credentials in the environment with the AWS CLI.

What we do need to figure out is why our session is 1 hour rather than the expected 6 hours.

**Comment by arsh** [comment]
One hour is the default. I think @sauraank was seeing this issue because he was not using `role-duration-seconds` to override the default.

**Comment by sauraank** [comment]
I understood that part and I have already added its fix in my PR #779 . I created this issue initially because I was thinking we would want the credentials to refreshed after 1 session and not having mountpoint failing. But, that is the expected behaviour. So, we can close this issue once my PR is merged.

**Comment by sauraank** [comment]
Closing this issue as resolved because #779  increases the session time for latency benchmark as 6 hours.
